src/crate/client/sqlalchemy/tests/compiler_test.py

```diff
@@ -74,25 +74,22 @@ def test_select_with_offset(self):
         """
         Verify the `CrateCompiler.limit_clause` method, with offset.
         """
-        self.metadata.bind = self.crate_engine
         selectable = self.mytable.select().offset(5)
-        statement = str(selectable.compile())
+        statement = str(selectable.compile(bind=self.crate_engine))
```

Comment:
Well, that's the same thing what I elaborated about at GH-490.

> "Implicit" and "Connectionless" execution, and "bound metadata", have been removed beginning with SQLAlchemy 2.0 [^1].

In this case, the use of "bound metadata" was not caught by turning on [`RemovedIn20Warnings`](https://docs.sqlalchemy.org/en/20/changelog/migration_20.html#migration-to-2-0-step-two-turn-on-removedin20warnings) within the preparation step GH-485, probably because the code is _directly_ setting the attribute(s) on the _existing_ instances here, without running `bind=` through any other regular constructor or initializer methods.

[^1]: https://docs.sqlalchemy.org/en/20/changelog/migration_20.html#implicit-and-connectionless-execution-bound-metadata-removed

Reply:
Diverted to GH-485.